src/fu-engine.c

```diff
@@ -2848,6 +2882,29 @@ fu_engine_update_detach (FuEngine *self, const gchar *device_id, GError **error)
 		return FALSE;
 	if (!fu_plugin_runner_update_detach (plugin, device, error))
 		return FALSE;
+
+	/* support older clients without the ability to do immediate requests */
+	if ((feature_flags & FWUPD_FEATURE_FLAG_REQUESTS) == 0 &&
+	    fu_device_get_request_cnt (device, FWUPD_REQUEST_KIND_IMMEDIATE) > 0) {
+
+		/* fallback to something sane */
+		if (fu_device_get_update_message (device) == NULL) {
+			g_autofree gchar *tmp = NULL;
+			tmp = g_strdup_printf ("Device %s needs to manually be put in update mode",
+					       fu_device_get_name (device));
+			fu_device_set_update_message (device, tmp);
```

Comment:
Is this true that all immediate requests are to go into update mode?  Or would there also be requests that are power cycle the device and the plugin continues to verify it?

Reply:
This is for `<screenshots><screenshot><caption>` which I think we specify specifically is for detach into bootloader. Perhaps we should just nuke the `Device %s needs to manually be put in update mode` prefix and put the entire string in the metadata?

Reply:
Yeah that's better

Reply:
Okay, removed.